Remove Java 8 support + move the testsuite from Travis to Azure

mstruk opened the pull request:
This PR implements https://github.com/strimzi/proposals/pull/151 for OAuth repo.

It also removes Java 8 support. The released Maven artifacts will now be built with Java 11.

ppatierno commented on .travis/build.sh:
```diff
@@ -11,6 +11,18 @@ exitIfError() {
+  echo "Usage: DISABLED=false ./build.sh"
+  exit 0
+fi
+
```
why aren't we just removing the `.travis` file? Didn't this proposal have the purpose of doing that? https://github.com/strimzi/proposals/blob/main/099-drop-travis-ci-and-testing-for-ppc-and-s390x.md

mstruk replied:
I can do that as well yes.

Frawless replied:
yes, `.travis` shoudl be removed.

mstruk commented:
Someone with permissions will have to go into project Settings and turn off Travis CI integration.
With `.travis/build.sh` gone the Travis CI check will now always fail.

ppatierno commented:
> Someone with permissions will have to go into project Settings and turn off Travis CI integration.
With .travis/build.sh gone the Travis CI check will now always fail.

I deleted the webhook which was notifying Travis CI but maybe it's too late for this PR and the Travis CI failure will stay?

Frawless commented on .azure/templates/jobs/build_java.yaml:
```diff
@@ -39,11 +36,34 @@ jobs:
-          MVN_ARGS: "-e -V -B"
+      - bash: "mvn ${MVN_ARGS} clean install -f examples/docker ${MVN_EXTRA_ARGS}"
+        displayName: "Test Examples"
+      - bash: |
+          cd examples/docker
+          ./spring/test-spring.sh
+        displayName: "Test Spring Example"
+        condition: and(succeeded(), eq(variables['jdk_version'], '17'))
```
Do we need this? What is the purpose of it?

mstruk replied:
The Spring Example test requires Java 17. The `succeeded()` function means that it will only execute if the previous step was successful (the compilation).

mstruk commented:
> I deleted the webhook which was notifying Travis CI but maybe it's too late for this PR and the Travis CI failure will stay? I have also removed the oauth repo from the Travis CI app (which is installed at global level within Strimzi org)

I can push a cosmetic change, and we'll see.

see-quick reviewed (approved):
LGTM. Thanks
